crates/turborepo-lib/src/process/child.rs

```diff
@@ -733,4 +736,47 @@ mod test {
         assert!(buffer == b"hello world\nhello moon\n" || buffer == b"hello moon\nhello world\n");
         assert_matches!(exit, Some(ChildExit::Finished(Some(0))));
     }
+
+    #[tokio::test]
+    async fn test_wait_with_with_non_utf8_output() {
+        let script = find_script_dir().join_component("hello_non_utf8.js");
+        let mut cmd = Command::new("node");
+        cmd.args([script.as_std_path()]);
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+        let mut child = Child::spawn(cmd, ShutdownStyle::Kill).unwrap();
+
+        let mut out = Vec::new();
+        let mut err = Vec::new();
+
+        let exit = child
+            .wait_with_piped_outputs(&mut out, &mut err)
+            .await
+            .unwrap();
+
+        assert_eq!(out, &[0, 159, 146, 150, b'\n']);
+        assert!(err.is_empty());
+        assert_matches!(exit, Some(ChildExit::Finished(Some(0))));
+    }
+
+    #[tokio::test]
+    async fn test_wait_with_non_utf8_single_outut() {
+        let script = find_script_dir().join_component("hello_non_utf8.js");
+        let mut cmd = Command::new("node");
+        cmd.args([script.as_std_path()]);
+        cmd.stdout(Stdio::piped());
+        cmd.stderr(Stdio::piped());
+        let mut child = Child::spawn(cmd, ShutdownStyle::Kill).unwrap();
+
+        let mut buffer = Vec::new();
+
+        let exit = child
+            .wait_with_single_piped_output(&mut buffer)
+            .await
+            .unwrap();
+
+        // There are no ordering guarantees so we accept either order of the logs
```

Comment:
I'm confused, what are the possible orders?

Reply:
Oops, got copy-pasta'd from another test.